### PR TITLE
[DiagnosticQol][SR-14505] Use DeclDescriptive kind in missing return data flow diagnostics

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Separate closure and declaration on diagnoseMissingReturn data flow diagnostics and use better descriptive naming.

Resolves SR-14505.

### CODE REVIEW

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

File: test/SILOptimizer/return.swift [hunk 1]
```diff
@@ -213,3 +213,39 @@ func testSR13753() {
...
+
+class SR14505_C {
+  static let a = false
+  let b = true
+
+  func method() -> Int {
+    if b {
+      return 0
+    }
+  } // expected-error {{missing return in a instance method expected to return 'Int'}}
```
Comment (harlanhaskins) [inline test/SILOptimizer/return.swift:244]: Should this say `an instance method` or remove the article altogether?

Comment (LucianoPAlmeida) [reply test/SILOptimizer/return.swift]: It make sense to remove the article, thanks @harlanhaskins :)

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Review by varungandhi-apple — APPROVED** [review]
Thanks for the fix! The diagnostic wording feels a little bit odd with the missing a/an, but that seems like somewhat tedious to get right, so I think this is good to go. 🚢

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test macOS

**Comment by LucianoPAlmeida** [comment]
@swift-ci please smoke test macOS

**Comment by LucianoPAlmeida** [comment]
"This build failed because of a Jenkins Error; typically a Java exception."
This seems unrelated, let's try it again ... 

@swift-ci please smoke test macOS

**Comment by LucianoPAlmeida** [comment]
@swift-ci please smoke test macOS

**Comment by varungandhi-apple** [comment]
That generally happens when the system is overloaded. Maybe we can retry after a few hours.

**Comment by LucianoPAlmeida** [comment]
> That generally happens when the system is overloaded.
Ah, I've seen this message couple of times, didn't know that. Thanks for the info @varungandhi-apple :)

**Pull request merged**